python/src/aiconfig/eval/metrics.py

```diff
@@ -67,3 +71,47 @@ def _calculate_brevity(output_datum: str) -> float:
         worst_value=float("inf"),
     ),
 )
+
+
+@dataclass
+class TextSentimentScores(CustomMetricValue):
+    mapping: dict[str, float]
+    pos: float
+    neg: float
+    neu: float
+    compound: float
+    highest: str
+
+
+def _get_sentiment_scores(output_datum: str) -> TextSentimentScores:
+    nltk.download("vader_lexicon", quiet=True)  # type: ignore
+    sid = SentimentIntensityAnalyzer()
+    mapping: dict[str, float] = sid.polarity_scores(output_datum)  # type: ignore
+    highest: str = pd.Series(mapping).idxmax()  # type: ignore
+    return TextSentimentScores(mapping=mapping, **mapping, highest=highest)
+
+
+def _get_sentiment(output_datum: str) -> str:
+    return _get_sentiment_scores(output_datum).highest
+
+
+sentiment_scores: Metric[str] = Metric(
+    evaluation_fn=_get_sentiment_scores,
+    metric_metadata=EvaluationMetricMetadata(
+        name="sentiment_scores",
+        description="Sentiment scores container object",
+        best_value=None,
+        worst_value=None,
+    ),
+)
+
+
+sentiment_class: Metric[str] = Metric(
+    evaluation_fn=_get_sentiment,
+    metric_metadata=EvaluationMetricMetadata(
+        name="sentiment_class",
+        description="top sentiment class",
+        best_value="pos",
+        worst_value="neg",
```

Comment:
There are two parts of the confusion here:
1. I added 2 different metrics, both of which are based on sentiment score, but they  have different types. `sentiment_scores` is a container that contains the mapping to floats. In contrast, this metric, `sentiment_class`, is just the string value of the highest-scoring class, i.e. `argmax_{classes} score(class)`.

2. There is a bug here, thanks for the catch: Since I defined best and worst for this metric, I have to define the ordering so that all values fall on `[worst, best]` and validation succeeds.